toc/rfc/rfc-draft-bindable-secrets.md

```diff
@@ -0,0 +1,293 @@
+  "relationships": {
+    "space": {
+      "data": {
+        "guid": "5a84d315-9513-4d74-95e5-f6a5501eeef7"
+      }
+    }
+  },
+  "metadata": {
+    "labels": {},
+    "annotations": {}
+  },
+  "links": {
+    "self": {
+      "href": "https://api.example.org/v3/service_instances/c89b3280-fe8d-4aa0-a42e-44465bb1c61c"
+    },
+    "space": {
+      "href": "https://api.example.org/v3/spaces/5a84d315-9513-4d74-95e5-f6a5501eeef7"
+    },
+    "secret_bindings": {
+      "href": "https://api.example.org/v3/secret_bindings?secret_guids=4f00b75b-3455-48f1-ba4a-f3b82ebad943"
+    },
+    "value": {
+      "href": "https://api.example.org/v3/secrets/4f00b75b-3455-48f1-ba4a-f3b82ebad943/value"
+    }
+  }
+}
+```
+
+A `secret` has a type that identifies its contents. To start we will support the `opaque` credential type (corresponds to the `json` [CredHub credential type](https://docs.cloudfoundry.org/credhub/credential-types.html)) which is designed to hold arbitrary user-defined data. In the future, this can be expanded to support more semantic secret types such as those [supported by Kubernetes](https://kubernetes.io/docs/concepts/configuration/secret/#secret-types) or other CredHub-supported types like `certificate`. `opaque` secrets consist of a JSON object whose keys map to files and values map to file content.
+
+#### Example `secret_binding` Object
+
+```json
+{
+  "guid": "dde5ad2a-d8f4-44dc-a56f-0452d744f1c3",
+  "created_at": "2015-11-13T17:02:56Z",
+  "updated_at": "2016-06-08T16:41:26Z",
+  "mount_path": "/etc/my-conf/",
+  "metadata": {
+    "annotations": {},
+    "labels": {}
+  },
+  "relationships": {
+    "app": {
+      "data": {
+        "guid": "74f7c078-0934-470f-9883-4fddss5b8f13"
+      }
+    },
+    "secret": {
+      "data": {
+        "guid": "4f00b75b-3455-48f1-ba4a-f3b82ebad943"
+      }
+    }
+  },
+  "links": {
+    "self": {
+      "href": "https://api.example.org/v3/service_credential_bindings/dde5ad2a-d8f4-44dc-a56f-0452d744f1c3"
+    },
+    "secret": {
+      "href": "https://api.example.org/v3/secrets/4f00b75b-3455-48f1-ba4a-f3b82ebad943"
+    },
+    "app": {
+      "href": "https://api.example.org/v3/apps/74f7c078-0934-470f-9883-4fddss5b8f13"
+    }
+  }
+}
+```
+
+The `mount_path` field defines the `tmpfs` directory that will contain files for the `secret`'s content. For example, consider an `opaque`-type `secret` with the following `value`:
+
+```json
+"value": {
+  "application.yml" : "---\napplication:\n ...",
+  "secret.txt": "my-secret"
+}
+```
+
+The above `secret` would mount files named `application.yml` and `secret.txt` at the path configured in the binding.
+
+### Creating a `secret`
+
+To create a `secret`, users will use a new CLI command: `cf create-secret`.
+
+```console
+cf create-secret SECRET_NAME -t opaque -p <CREDENTIALS_INLINE>
+```
+
+```console
+cf create-secret SECRET_NAME -t opaque -f <FILE_PATH_TO_CREDENTIALS_FILE>
+```
+
+* `-t` the secret `type`, defaults to `opaque`
```

Comment:
If we do not have a different type right now is it worth it to add this flag?

Reply:
Yeah, we don't need it for now. I'll update this and remove the flag.